src/libs/ajax/AzureStorage.ts

```diff
@@ -111,6 +111,54 @@ export const AzureStorage = (signal?: AbortSignal) => ({
     }), notebooks)
   },
 
+  blobMetadata: (azureStorageUrl: string) => {
+    const getObjectMetadata = async () => {
+      const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
+      const fileName = _.last(azureStorageUrl?.split('/'))?.split('.')?.join('.')
+
+      try {
+        // assumption is made that container name guid in uri always matches the workspace Id guid it is present in
```

Comment:
I think this comment would be more useful on the line that parses the workspace ID from the URL.
```ts
const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
```